### DISCUSSION TITLE
Nextjs13 just killed SWR?

### DISCUSSION THREAD

**Discussion by osseonews** [vercel/swr#2223] [Q&A]
Reading the Nextjs13 docs, and it says using SWR could hurt performance and Fetch is not allowed in client components? What? Vercel has a team building out this incredible library, and now they are basically saying it has no future. Yeah, you don't have to use the app folder in NextJs13 right now if you don't want to, but it's clear what direction NextJs is heading in. So if you build on SWR you will have to refactor everything eventually once the app folder becomes standard.  Why did Vercel do this? What was wrong with SWR exactly? Perfectly solved data fetching issues client side.

**Comment by raulmarindev** [comment]
> Reading the Nextjs13 docs, and it says using SWR could hurt performance and Fetch is not allowed in client components? What? Vercel has a team building out this incredible library, and now they are basically saying it has no future. Yeah, you don't have to use the app folder in NextJs13 right now if you don't want to, but it's clear what direction NextJs is heading in. So if you build on SWR you will have to refactor everything eventually once the app folder becomes standard.  Why did Vercel do this? What was wrong with SWR exactly? Perfectly solved data fetching issues client side. 

I believe this is what you are referring to: https://beta.nextjs.org/docs/data-fetching/fundamentals

>Note:
>*fetch is currently not supported in Client Components. Although it is possible to fetch data in Client Components by, for example, using a third-party library such as SWR, you might run into performance issues when doing so. When possible, try to keep your data fetching logic inside a Server Component using use and/or fetch, and pass this data down to its Client children components.

>*It's not possible to pass data between a parent layout and its children. However, you can fetch the same data in a route more than once and React will automatically dedupe the requests without affecting performance.

I agree a bit of clarification regarding this would be most welcome.

**Reply by osseonews** [reply]
Read the documents a bit more. Of course, you don't have to use the "App" folder, as it's just beta, and then everything remains the same. But, Next is heading in the direction of the App folder. And if you use the App folder, then there is a concept of static and dynamic fetching. Both will cache and dedupe your fetch request. So I don't see what the future of SWR will be in you move an App folder. It just won't be necessary, as the entire point is to ultimately do all fetchdata server side using the new data fetching model.  I guess there is some stuff you may want to do on the client, but even then, it will be complicated to implement (see Fetch and Use in Client Components in the docs).

**Reply by raulmarindev** [reply]
Yep, I believe you're on point.

**Comment by ymc9** [comment]
But there'll always be needs for client-side fetching, right? After a page is loaded and interactive. So will SWR be mostly just for that scenario in the future?

Or is it the case that Server Components will eventually completely eliminate the need of client-side fetching?

**Comment by stunaz** [comment]
I had the same questions. and it is also my understanding that swr is no longer needed when using the app folder.

**Comment by ghost** [comment]
There's always a need for client-side fetching
